stacks/towerofhanoi.js

```diff
@@ -0,0 +1,53 @@
+const Stack = require("./Stack");
+
+function TowerOfHanoi (discCount) {
+    this.discCount = discCount;
+    this.towerA = new Stack(discCount);
+    this.towerB = new Stack(discCount);
+    this.towerC = new Stack(discCount);
+
+    var noOfDiscs = discCount;
+    while (noOfDiscs-- > 0) {
+        this.towerA.push("Disc");
+    }
+}
+
+TowerOfHanoi.prototype.startMovingDiscs = function () {
+
+    this.solvePuzzle (this.discCount, "A", "C", "B");
+}
+
+TowerOfHanoi.prototype.solvePuzzle = function (discCount, source, destination, buffer) {
+
+    if (discCount >= 1) {
+
+        this.solvePuzzle(discCount-1, source, buffer, destination);
+
+        var poppedElement = "";
+        if (source === "A") {
```

Comment:
if else makes it a little difficult to understand. Get this function to just have stacks as three inputs instead of identifier a, b and c